pkg/project/admission/lifecycle/admission.go

```diff
@@ -140,10 +141,26 @@ func (e *lifecycle) Handles(operation admission.Operation) bool {
 }
 
 func NewLifecycle(client client.Interface, creatableResources sets.String) (admission.Interface, error) {
```

Comment:
For example, kube has:

```
// Interface is an abstract, pluggable interface for Admission Control decisions.
type Interface interface {
	// Admit makes an admission decision based on the request attributes
	Admit(a Attributes) (err error)

	// Handles returns true if this admission controller can handle the given operation
	// where operation can be one of CREATE, UPDATE, DELETE, or CONNECT
	Handles(operation Operation) bool
}
```

And origin could have:
```
// Interface is an abstract, pluggable interface for Admission Control decisions.
type Interface interface {
	// Configure performs origin specific configuration
	Configure(origin client) (err error)
}
```

And we can use reflection to check if an `admission.Interface` implemented the corresponding `origin.Interface` and do an additional `Configure` call on those handlers so information can be passed explicilty.

Hope that makes sense.